GliaWidgets/Sources/PushNotifications/PushNotifications.swift

```diff
@@ -15,17 +24,95 @@ public struct PushNotifications {
         )
     }
 
-    /// Set the current handler that the SDK is forwarding the
-    /// UNNotificationResponse.actionIdentifier to.
+    /// Notifies the push notifications system that the application failed to register for remote notifications.
+    ///
+    /// This method is typically called from the `UIApplicationDelegate`'s failure callback.
+    ///
+    /// - Parameters:
+    ///   - application: The application instance that attempted registration.
+    ///   - error: The error encountered during the registration process.
+    ///
+    public func applicationDidFailToRegisterForRemoteNotificationsWithError(
+        application: UIApplication,
+        error: any Error
+    ) {
+        environment.coreSdk.pushNotifications.applicationDidFailToRegisterForRemoteNotificationsWithError(
+            application,
+            error
+        )
+    }
+
+    /// Sets the current push action handler that the SDK uses to forward notification actions.
+    ///
+    /// This handler is executed in response to user interactions with notifications (via
+    /// `UNNotificationResponse.actionIdentifier`).
+    ///
+    /// - Parameter handler: An optional `PushActionBlock` closure used as a callback for push actions.
+    ///
     public func setPushHandler(_ handler: GliaCoreSDK.PushActionBlock?) {
         environment.coreSdk.pushNotifications.setPushHandler(handler)
     }
 
-    /// The current handler that the SDK is forwarding the
-    /// UNNotificationResponse.actionIdentifier to.
+    /// Retrieves the current push action handler.
+    ///
+    /// - Returns: An optional `PushActionBlock` closure that is currently set to handle push notification actions.
+    ///
     public func pushHandler() -> GliaCoreSDK.PushActionBlock? {
```

Comment:
Do we need to expose `pushHandler` getter?

Reply:
You're right, I removed the getter [500bf67](https://github.com/salemove/ios-sdk-widgets/pull/1266/commits/500bf67d08536f45b6eb7bea8f0d78704626cc5e)